File: caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main (int argc, string argv[])
{
    //Checking for 2 arguments in the command line
    if (argc == 2 && argv[2] >= 0)
    {
        //Convert string to an integer
        int k = atoi(argv[1]);

        // Prompt user for input
        string plaintext = get_string("plaintext: ");

        //Encrypting
        printf("ciphertext: ");

        for (int i = 0, n = strlen(plaintext); i < n; i++)
        {
            //Checking if a character is an alphabetical letter
            if (isalpha(plaintext[i]))
            {
                //Checking the case of the letter
                if (islower(plaintext[i]))
                {
                    printf("%c", (97 + ((plaintext[i] - 97 + k) % 26)));
                }
                if (isupper(plaintext[i]))
                {
                    printf("%c", (65 + ((plaintext[i] - 65 + k) % 26)));
                }
            }
            else
            {
                printf("%c", plaintext[i]);
            }
        }
        printf("\n");
    }
    else
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}

File: caesar.py
import sys
from cs50 import get_string

# Checking for 2 arguments in the command line
if len(sys.argv) == 2:

    # Convert string to an integer
    k = int(sys.argv[1])
    if k < 0:
        print("k should be non-negative")
        exit(1)

    # Prompt user for input
    plaintext = get_string("plaintext: ")

    # Encrypting
    print("ciphertext: ", end="")

    for c in plaintext:
        if c.isalpha():
            if c.islower():
                c = chr(ord('a') + (ord(c) - ord('a') + k) % 26)
                print(c, end="")
            if c.isupper():
                c = chr(ord('A') + (ord(c) - ord('A') + k) % 26)
                print(c, end="")
        else:
            print(c, end="")
    print()
else:
    print("Usage: python caesar.py k")
    exit(1)
